### PR TITLE
fixed a bug where error tiles growing over time do not return to norm…

### PR DESCRIPTION
**Pull request by MollocH**

### CODE REVIEW

File: src/Enum/MonitoringStatus.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+namespace App\Enum;
+
+use MyCLabs\Enum\Enum;
+
+class MonitoringStatus extends Enum
+{
+    private const OK = 'ok';
+    private const ERROR = 'error';
+
+    public static function OK(): MonitoringStatus
```
Comment (cawolf) [inline src/Enum/MonitoringStatus.php]: Named Constructors sollten automatisch von der Bibliothek bereitgestellt sein, kann weg

File: src/Enum/MonitoringStatus.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    {
+        return new MonitoringStatus(self::OK);
+    }
+
+    public static function ERROR(): MonitoringStatus
```
Comment (cawolf) [inline src/Enum/MonitoringStatus.php]: same

File: tests/Unit/Service/Persistance/PriorityServiceTest.php [hunk 3]
```diff
@@ -26,39 +26,108 @@ public function setUp()
...
+        int $expectedPriority
+    ): void {
+        $this->monitoringDataRepository->find('id')->shouldBeCalledOnce()->willReturn($oldMonitoringData);
 
-        self::assertSame(1, $this->subject->calculate($monitoringDataDto));
+        self::assertSame($expectedPriority, $this->subject->calculate($newMonitoringData));
     }
 
-    public function testCalculateSameOrLowerPriority(): void
+    public function provideMonitoringData()
```
Comment (cawolf) [inline tests/Unit/Service/Persistance/PriorityServiceTest.php]: `: array`

File: tests/Unit/Service/Persistance/PriorityServiceTest.php [hunk 4]
```diff
@@ -26,39 +26,108 @@ public function setUp()
...
+            'no old monitoring | new monitoring ok' => [
+                $newMonitoringData = $this->createMonitoringDataDto('ok', 10),
+                null,
+                10
+            ],
+        ];
     }
 
-    public function testCalculate(): void
+    private function createMonitoringDataDto(string $status, int $priority)
```
Comment (cawolf) [inline tests/Unit/Service/Persistance/PriorityServiceTest.php]: `: MonitoringDataDto`

File: tests/Unit/Service/Persistance/PriorityServiceTest.php [hunk 5]
```diff
@@ -26,39 +26,108 @@ public function setUp()
...
+            '* 2'
         );
-        $monitoringData = new MonitoringDataDocument(
-            'id', 'satus', new DateTime('2019-01-01 00:00:00'), 'payload', 27, 60, new DateTimeImmutable(), 'root.branch.leaf', 5, '* 2'
-        );
-        $this->monitoringDataRepository->find('id')->shouldBeCalledOnce()->willReturn($monitoringData);
+    }
 
-        self::assertSame(27, $this->subject->calculate($monitoringDataDto));
+    private function createMonitoringDataDocument(string $status, int $priority)
```
Comment (cawolf) [inline tests/Unit/Service/Persistance/PriorityServiceTest.php]: `: MonitoringDataDto`

File: tests/Unit/Service/Persistance/PriorityServiceTest.php [hunk 6]
```diff
@@ -26,39 +26,108 @@ public function setUp()
...
+                $newMonitoringData = $this->createMonitoringDataDto('error', 20),
+                $oldMonitoringData = $this->createMonitoringDataDocument('error', 20),
+                20
+            ],
+            'old monitoring prio < new monitoring prio (both error)' => [
+                $newMonitoringData = $this->createMonitoringDataDto('error', 20),
+                $oldMonitoringData = $this->createMonitoringDataDocument('error', 10),
+                20
+            ],
+            'old monitoring prio OK > new monitoring prio ERROR' => [
```
Comment (cawolf) [inline tests/Unit/Service/Persistance/PriorityServiceTest.php:60]: Hm, den Fall hätte ich anders erwartet. Der neue Push sollte die neue Prio setzen (genau wie im Beispiel direkt hiervor), oder nicht?

**Review by cawolf — CHANGES_REQUESTED** [review]

File: src/Enum/MonitoringStatus.php (hunk 1 above)
Comment (MollocH) [reply src/Enum/MonitoringStatus.php]: Ja werden sie, allerdings meckert dann die IDE rum. Deshalb habe ich sie explizit erstellt.

File: tests/Unit/Service/Persistance/PriorityServiceTest.php (hunk 6 above)
Comment (MollocH) [reply tests/Unit/Service/Persistance/PriorityServiceTest.php:60]: Das ist der Theorie nach ein unmöglicher Fall. Es sei denn du hättest ursprünglich mit 20 gespusht und würdest jetzt über die API mit 10 pushen.

Wenn dich das stört kann ich
```
 && $oldMonitoringDataDocument->getStatus() === MonitoringStatus::ERROR()->getValue()
``` 
hinzufügen.

Dann verändert sich genau nur dieses Verhalten laut Tests.

Comment (cawolf) [reply tests/Unit/Service/Persistance/PriorityServiceTest.php:60]: Hm, ok, eigentlich ist das ein Edge Case. Könnte nur passieren, wenn jemand den Pusher umkonfiguriert, während das Monitoring in `error` ist.
Würde ich so lassen, passt schon.

File: src/Enum/MonitoringStatus.php (hunk 1 above)
Comment (cawolf) [reply src/Enum/MonitoringStatus.php]: Ja, gut, aber da hat die IDE halt Pech. Wenn wir jedes mal für magic functions Platzhalter für die IDE erstellen müssten, wäre der ganze Sinn der magic functions ja hinüber.

**Review by cawolf — APPROVED** [review]
